Jenkinsfile

```diff
@@ -10,116 +39,149 @@ pipeline {
     buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
     timestamps()
   }
+  parameters {
+    string(name: "NewRelease", defaultValue: "${NewRelease}", description: "Your Current Release is: ${LastRelease} Change the version of New Release if necessary")
+    string(name: 'ECRURI', defaultValue: '054017840000.dkr.ecr.us-east-1.amazonaws.com', description: 'Enter the URI of the Container Registry')
+    string(name: 'Email', defaultValue: 'vecinomio@gmail.com', description: 'Enter the desired Email for the Job notifications')
+    choice(
+      choices: [
+        'us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'ap-east-1', 'ap-south-1',
+        'ap-southeast-1', 'ap-southeast-2', 'ap-northeast-1', 'ap-northeast-2', 'ap-northeast-3',
+        'ca-central-1', 'cn-north-1', 'cn-northwest-1', 'eu-central-1', 'eu-west-1', 'eu-west-2',
+        'eu-west-3', 'eu-north-1', 'me-south-1', 'sa-east-1'
+      ], name: 'AWSRegion', description: 'Choose the desired AWS region'
+    )
+    booleanParam(name: 'Build', defaultValue: true, description: 'Specify to Build App and do Tests')
+    booleanParam(name: 'Release', defaultValue: false, description: 'Specify to deliver an artifact to ECR and tags to Github repos')
+    booleanParam(name: 'Deployment', defaultValue: false, description: 'Specify to Deploy a new version of App to ECS')
+    choice(name: 'NewVersionTrafficWeight', choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: 'Choose amount of traffic to the new vesion of the App')
+  }
   environment {
-    ECRURI = '054017840000.dkr.ecr.us-east-1.amazonaws.com'
+    ECRURI = "${params.ECRURI}"
+    AWSRegion = "${params.AWSRegion}"
     AppRepoName = 'snakes'
-    OPSRepoURL = 'https://github.com/IYermakov/DevOpsA3Training.git'
-    OPSRepoBranch = 'ecs-snakes'
-    Tag = ""
-    Email = 'vecinomio@gmail.com'
-    DelUnusedImage = 'docker image prune -af --filter="label=maintainer=devopsa3"'
-
+    OPSRepoURL = 'git@github.com:IYermakov/DevOpsA3Training.git'
+    OPSRepoBranch = 'master'
+    BuildAndTest = "${params.Build}"
+    Release = "${params.Release}"
+    Deployment = "${params.Deployment}"
+    Tag = "${params.NewRelease}"
+    CurrentVersionTrafficWeight = (10 - "${params.NewVersionTrafficWeight}".toInteger()).toString()
```

Comment:
would extract magic constant `10` to variable

Reply:
Done